lib/core/request.js

```diff
@@ -143,7 +143,7 @@ class Request {
       ? method === 'HEAD' || method === 'GET'
       : idempotent
 
-    this.blocking = blocking == null ? false : blocking
+    this.blocking = blocking ?? this.method !== 'HEAD'
```

Comment:
We could also check range request headers and/or range and/or content-length response header to dynamically determine this.